### ISSUE TITLE
[a11y]: MultiSelect - "Role and State of the control is not announced by the screen reader."

### ISSUE THREAD

**Issue by szamoca-at-the-corp** [carbon-design-system/carbon#20890]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.81.0

### React version

v18.3.1

### Automated testing tool and ruleset

JAWS_Firefox

### Description

**Origin**

Our application received the following issue discovery from the AVT team:
> Component: Multi selection box
>
> Issue description: Role and State of the control is not announced by the screen reader. When focus is on multi selection options inside the combo box.
>
> Recommended fix: Role and State of the control should be announced by the screen reader. So that user will get to know that the focused controls is of which type and that control is in which state.
>
> User impact: People using a screen reader, including blind, low vision, and neurodivergent people will be impacted when appropriate Name, Role and State information is not announced by the screen reader.
>
> WCAG Mapping: 4.1.2 Name, Role, Value
>
> Level of Compliance: A
>
> Severity: Medium
>
> Test tool: JAWS_Firefox

**Dev context**

- We are using Carbon's `MultiSelect` component
- Amongst the props, we pass all the required ones: `id`, `items`, `label`
- Component change handler is a simple React `setState` function
- As items, we pass a filtered array of simple strings

### WCAG 2.1 Violation

Success Criterion 4.1.2 Name, Role, Value

### Reproduction/example

https://stackblitz.com/edit/github-jdymvwuu?file=src%2FExample%2FChildrenContent.jsx

### Steps to reproduce

1. Open side panel
2. Open dropdown and move through items using the arrow keys
3. Select and unselect items

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
Hey @szamoca-at-the-corp  - just wanted to confirm how are you using MacOS and JAWS combination ? 
I just tested this on MacOS with VoiceOver and it is working as expected

**Comment by szamoca-at-the-corp** [comment]
@2nikhiltom Thank you for looking into this. And actually I don't know what the test team's setup is unfortunately. We received this ticket with the last round of AVT, and we only know the context that's in the quoted section above. Do you think it is a false positive issue?

**Comment by 2nikhiltom** [comment]
verified this on windows machine on JAWS, this looks like a valid bug
